### PR TITLE
migration: modules typescript 6

### PR DESCRIPTION
**Pull request by bang9**
[CLNP-1247](https://sendbird.atlassian.net/browse/CLNP-1247?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ)

Modules

- ./modules/Channel/context/hooks/useDeleteMessageCallback.js: useDeleteMessageCallback.js
- ./modules/Channel/context/hooks/useGetChannel.js: useGetChannel.js
- ./modules/Channel/context/hooks/useResendMessageCallback.js: useResendMessageCallback.js
- ./modules/Channel/context/hooks/useScrollDownCallback.js: useScrollDownCallback.js
- ./modules/Channel/context/hooks/useSendFileMessageCallback.js: useSendFileMessageCallback.js
- ./modules/Channel/context/hooks/useSendMessageCallback.js: useSendMessageCallback.js
- ./modules/Channel/context/hooks/useToggleReactionCallback.js: useToggleReactionCallback.js
- ./modules/Channel/context/hooks/useUpdateMessageCallback.js: useUpdateMessageCallback.js
- (removed) ./modules/Channel/components/FileViewer/types.js: types.js

[CLNP-1247]: https://sendbird.atlassian.net/browse/CLNP-1247?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/Channel/context/ChannelProvider.tsx [hunk 1]
```diff
@@ -140,7 +140,7 @@ interface ChannelProviderInterface extends ChannelContextProps, MessageStoreInte
   isScrolled?: boolean;
   setIsScrolled?: React.Dispatch<React.SetStateAction<boolean>>;
   messageActionTypes: typeof channelActions;
-  messagesDispatcher: CustomUseReducerDispatcher;
+  messagesDispatcher: React.Dispatch<ChannelActionTypes>;
```
Comment (liamcho) [inline src/modules/Channel/context/ChannelProvider.tsx:143]: don't we need to update index.d.ts file accordingly?

Comment (bang9) [reply src/modules/Channel/context/ChannelProvider.tsx:143]: Yes, we will soon generate type definition through ts source code, so I don't think we have to change it.

**Review by liamcho — APPROVED** [review]
LGTM thank you for the hard work @bang9 !

File: src/modules/Channel/context/hooks/useHandleChannelEvents.ts [hunk 2]
```diff
@@ -53,15 +54,15 @@ function useHandleChannelEvents({
...
+    ?.find((feature) => (feature === DELIVERY_RECEIPT));
 
   useEffect(() => {
     const channelUrl = currentGroupChannel?.url;
     const channelHandlerId = uuidv4();
     if (channelUrl && sdkInit) {
       const channelHandler: GroupChannelHandler = {
         onMessageReceived: (channel, message) => {
-          if (compareIds(channel?.url, channelUrl)) {
+          if (channel.isGroupChannel() && compareIds(channel?.url, channelUrl)) {
```
Comment (HoonBaek) [inline src/modules/Channel/context/hooks/useHandleChannelEvents.ts:65]: 👍

**Review by HoonBaek — APPROVED** [review]
LGTM

**Pull request merged**